### PR TITLE
ci(release): use new set of credentials for GPG and Maven Central 

### PR DESCRIPTION
**Pull request by v1v**
### What

Use the Vault secrets accessed to run the Release/Snapshots.

Those secrets are managed by the Release team and they provided specific ones for this project.

### Why

The release team requested to use them.

### CODE REVIEW

**Review by reakaleek — DISMISSED** [review]

File: .buildkite/hooks/pre-command [hunk 1]
```diff
@@ -34,20 +19,22 @@ mkdir -p $GNUPGHOME
...
-SERVER_PASSWORD=$(vault read -field password secret/release/nexus)
+SERVER_PASSWORD=$(vault read -field password $NEXUS_SECRET)
 export SERVER_PASSWORD
 
 # Signing keys
-vault read -field=key secret/release/signing >$KEY_FILE
-KEYPASS_SECRET=$(vault read -field=passphrase secret/release/signing)
+GPG_SECRET=kv/data/ci-shared/release-eng/team-release-secrets/ecs-logging/gpg
+vault read -field="keyring" $GPG_SECRET | base64 -d > $KEY_FILE
+## NOTE: passphase is the name of the field.
```
Comment (reakaleek) [inline .buildkite/hooks/pre-command:32]:
```suggestion
## NOTE: This is not a typoe. "passphase" is the name of the field.
```

**Review by JonasKunz — DISMISSED** [review]

**Comment by v1v** [comment]
On hold for now. https://github.com/elastic/elastic-otel-java/pull/71 was merged, but there are some missing configurations when accessing the Vault secrets, so we are waiting for feedback from the system owners.

File: .buildkite/hooks/pre-command [hunk 2]
```diff
@@ -34,20 +19,22 @@ mkdir -p $GNUPGHOME
 chmod -R 700 $TMP_WORKSPACE
 
 echo "--- Prepare keys context :key:"
-VAULT_TOKEN=$(vault write -field=token auth/approle/login role_id="$VAULT_ROLE_ID_SECRET" secret_id="$VAULT_SECRET_ID_SECRET")
-export VAULT_TOKEN
-
 # Nexus credentials
-SERVER_USERNAME=$(vault read -field username secret/release/nexus)
+NEXUS_SECRET=kv/ci-shared/release-eng/team-release-secrets/ecs-logging/maven_central
+SERVER_USERNAME=$(vault read -field username $NEXUS_SECRET)
```
Comment (v1v) [inline .buildkite/hooks/pre-command:24]:
```suggestion
SERVER_USERNAME=$(vault kv get --field="username" $NEXUS_SECRET)
```

File: .buildkite/hooks/pre-command [hunk 3]
```diff
@@ -34,20 +19,22 @@ mkdir -p $GNUPGHOME
...
 export SERVER_USERNAME
-SERVER_PASSWORD=$(vault read -field password secret/release/nexus)
+SERVER_PASSWORD=$(vault read -field password $NEXUS_SECRET)
```
Comment (v1v) [inline .buildkite/hooks/pre-command:26]:
```suggestion
SERVER_PASSWORD=$(vault kv get --field="password" $NEXUS_SECRET)
```

File: .buildkite/hooks/pre-command [hunk 4]
```diff
@@ -34,20 +19,22 @@ mkdir -p $GNUPGHOME
...
 export SERVER_PASSWORD
 
 # Signing keys
-vault read -field=key secret/release/signing >$KEY_FILE
-KEYPASS_SECRET=$(vault read -field=passphrase secret/release/signing)
+GPG_SECRET=kv/data/ci-shared/release-eng/team-release-secrets/ecs-logging/gpg
+vault read -field="keyring" $GPG_SECRET | base64 -d > $KEY_FILE
+## NOTE: This is not a typo. "passphase" is the name of the field.
+KEYPASS_SECRET=$(vault read -field="passphase" $GPG_SECRET)
```
Comment (v1v) [inline .buildkite/hooks/pre-command:33]:
```suggestion
KEYPASS_SECRET=$(vault kv get --field="passphase" $GPG_SECRET)
```

File: .buildkite/hooks/pre-command [hunk 5]
```diff
@@ -34,20 +19,22 @@ mkdir -p $GNUPGHOME
...
 export KEYPASS_SECRET
-export KEY_ID_SECRET=D88E42B4
+KEY_ID=$(vault kv get --field="key_id" $GPG_SECRET)
```
Comment (v1v) [inline .buildkite/hooks/pre-command:35]:
```suggestion
KEY_ID=$(vault kv get --field="key_id" $GPG_SECRET)
```

File: .buildkite/hooks/pre-command [hunk 6]
```diff
@@ -34,20 +19,22 @@ mkdir -p $GNUPGHOME
...
 export SERVER_USERNAME
-SERVER_PASSWORD=$(vault read -field password secret/release/nexus)
+SERVER_PASSWORD=$(vault kv get --field="password" $NEXUS_SECRET)
 export SERVER_PASSWORD
 
 # Signing keys
-vault read -field=key secret/release/signing >$KEY_FILE
-KEYPASS_SECRET=$(vault read -field=passphrase secret/release/signing)
+GPG_SECRET=kv/data/ci-shared/release-eng/team-release-secrets/ecs-logging/gpg
+vault read -field="keyring" $GPG_SECRET | base64 -d > $KEY_FILE
```
Comment (v1v) [inline .buildkite/hooks/pre-command:31]:
```suggestion
vault kv get --field="keyring" $GPG_SECRET | base64 -d > $KEY_FILE
```

File: .buildkite/hooks/pre-command [hunk 7]
```diff
@@ -34,20 +19,22 @@ mkdir -p $GNUPGHOME
...
+SERVER_USERNAME=$(vault kv get --field="username" $NEXUS_SECRET)
 export SERVER_USERNAME
-SERVER_PASSWORD=$(vault read -field password secret/release/nexus)
+SERVER_PASSWORD=$(vault kv get --field="password" $NEXUS_SECRET)
 export SERVER_PASSWORD
 
 # Signing keys
-vault read -field=key secret/release/signing >$KEY_FILE
-KEYPASS_SECRET=$(vault read -field=passphrase secret/release/signing)
+GPG_SECRET=kv/data/ci-shared/release-eng/team-release-secrets/ecs-logging/gpg
```
Comment (v1v) [inline .buildkite/hooks/pre-command:30]:
```suggestion
GPG_SECRET=kv/ci-shared/release-eng/team-release-secrets/ecs-logging/gpg
```

**Review by reakaleek — APPROVED** [review]

**Pull request merged**